stonesoup/tracker/pointprocess.py

```diff
@@ -0,0 +1,184 @@
+                            self.gaussian_mixture.components,
+                            detections,
+                            time
+                            )
+            # Perform GM Update
+            self.gaussian_mixture = self.updater.update(hypotheses)
+            # Reduce mixture - Pruning and Merging
+            self.gaussian_mixture.components = \
+                self.reducer.reduce(self.gaussian_mixture.components)
+            # Update the tracks
+            self.tracks()
+            yield time, self.target_tracks
+
+    def end_tracks(self):
+        """
+        Ends the tracks (:class:`Track`) that do not have an associated
+        component within the filter.
+
+        Parameters
+        ==========
+        self : :state:`GaussianMixtureMultiTargetTracker`
+            Current GM Multi Target Tracker at time :math:`k`
+        """
+        component_tags = [component.tag for component in self.gaussian_mixture]
+        for tag in self.target_tracks:
+            if tag not in component_tags:
+                # Track doesn't have matching component, so end
+                self.target_tracks[tag].active = False
+
+    @property
+    def extracted_target_states(self):
+        """
+        Extract all target states from the Gaussian Mixture that
+        are above an extraction threshold.
+        """
+        if self.gaussian_mixture:
+            extracted_states = \
+                [x for x in self.gaussian_mixture if
+                 x.weight > self.extraction_threshold]
+        else:
+            extracted_states = []
+        return extracted_states
+
+    @property
+    def estimated_number_of_targets(self):
+        """
+        The number of hypothesised targets.
+        """
+        if self.gaussian_mixture:
+            estimated_number_of_targets = sum(component.weight for component in
+                                              self.gaussian_mixture)
+        else:
+            estimated_number_of_targets = 0
+        return estimated_number_of_targets
+
+
+class GMPHDTargetTracker(PointProcessMultiTargetTracker):
+    """
+    A implementation of the Gaussian Mixture
+    Probability Hypothesis Density (GM-PHD) multi-target filter
+
+    References
+    ----------
+
+    .. [1] B.-N. Vo and W.-K. Ma, “The Gaussian Mixture Probability Hypothesis
+    Density Filter,” Signal Processing,IEEE Transactions on, vol. 54, no. 11,
+    pp. 4091–4104, 2006..
+    """
+
+
+class GMLCCTargetTracker(PointProcessMultiTargetTracker):
+    """
+    A implementation of the Gaussian Mixture
+    Linear Complexity with Cumulants (GM-LCC) multi-target filter
+
+    References
+    ----------
+
+    .. [1] D. E. Clark and F. De Melo. “A Linear-Complexity Second-Order
+        Multi-Object Filter via Factorial Cumulants”. In: 2018
+        21st International Conference on Information Fusion (FUSION).
+        2018.
+        DOI: 10.23919/ICIF.2018.8455331. ..
+    """
```

Comment:
Do we really need separate classes for GMPHD and GMLCC? Right now there is no difference between these classes and the PointProcessMultiTargetTracker class. If the only difference is determined by what objects are passed in on creation e.g. Detector, Hypothesizer, Update, Predictor etc. then we may want to capture that somewhere else.

Reply:
I agree so I've removed the tracker subclasses. I had them there as placeholders in case there were any differences. The differences in the filters are now captured in the Updater subclasses